added intership service, route and unittest for it

louisbank-edu-thailand opened the pull request.

louisbank-edu-thailand committed:
added intership service, route and unittest

mythicalbadger commented on src/app/internship/routes.py:
```diff
@@ -1,8 +1,35 @@
+        company, position, website, deadline, author_id, time_period_id
+    )
+
+    response = {"message": "Time period added successfully"}
```
Should return "Internship created successfully" I think

louisbank-edu-thailand committed:
added view all internships endpoint

louisbank-edu-thailand committed:
Merge branch 'main' into user_can_create_internship_redundant_second_branch

louisbank-edu-thailand committed:
more prepop routes and endpoint

added some more prepopulation stuff to have an internship in the database without foreign key problems. also added get_all internships endpoint. might have to restart database with volume rm ... so that foreign id´s fit

louisbank-edu-thailand committed:
minor changes to create internship

mythicalbadger reviewed (approved):
LGTM

mythicalbadger commented on src/app/internship/routes.py:
```diff
@@ -1,8 +1,66 @@
+    return make_response(jsonify(response), 201)
+
+
+@bp.route("/internship/get_internships", methods=["GET"])
```
Should be `/internships` by our OpenAPI

mythicalbadger commented on tests/test_internship.py:
```diff
@@ -0,0 +1,47 @@
+
+def test_get_internships(test_client: FlaskClient) -> None:
+    """Test the get internships endpoint."""
+    response = test_client.get("/internship/get_internships")
```
Should be `/internships` by our OpenAPI

mythicalbadger committed:
fix: /internships endpoint

mythicalbadger committed:
fix: fixed wrong url lmao
